### PR TITLE
Optimize(eos_designs): Move upgrade into eos_designs role and optimize it a bit

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Move upgrade into eos_designs role and optimize it a bit

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

- Move upgrade functionality into `eos_designs` role itself and remove `upgrade_tools` role.
- Updated README.md for `eos_designs`.
- Minor modifications like path name for upgrade - in preparation to have a similar upgrade functionality for `eos_cli_config_gen`

## How to test

- Updated molecule upgrade scenarios to use the new method.
- No configuration changes caused by this.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/templates/upgrade/upgrade-2.x-to-3.0/l3leaf.j2 [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+# Retain 2.x behavior by setting spine.bgp_as on peer-node_groups
+    structured_config:
+      router_bgp:
+        peer_groups:
+{%         if underlay_routing_protocol | arista.avd.default('ebgp') | lower in ['bgp','ebgp'] %}
+            {{ bgp_peer_groups.IPv4_UNDERLAY_PEERS.name | arista.avd.default("IPv4-UNDERLAY-PEERS") }}:
+              remote_as: {{ spine.bgp_as }}
+{%         endif %}
+            {{ bgp_peer_groups.EVPN_OVERLAY_PEERS.name | arista.avd.default("EVPN-OVERLAY-PEERS") }}:
+              remote_as: {{ spine.bgp_as }}
```
Comment (xaviramon) [inline ansible_collections/arista/avd/roles/eos_designs/templates/upgrade/upgrade-2.x-to-3.0/l3leaf.j2]: Typo. Generated with wrong identation:
```
# Retain 2.x behavior by setting spine.bgp_as on peer-node_groups
    structured_config:
      router_bgp:
        peer_groups:
            IPv4-UNDERLAY-PEERS:
              remote_as: 65100
            EVPN-OVERLAY-PEERS:
              remote_as: 65100
```

**Comment by xaviramon** [comment]
I believe there are some changes that could be implemented to optimise actions and also to provide future upgrades.

1. When executing the playbook as instructed, this generates the folder with the changes but also executes the eos_designs module. This does not generate the configuration in EOS style, unless the user defines the playbook. I believe it makes sense to only generate the upgrade folder without executing the eos_designs role or provide a playbook that generates the config, so device configuration can be compared.
2. I think we should version the tasks so we can reuse the role for future upgrades.

**Comment by ClausHolbechArista** [comment]
@xaviramon ready for new review. I hope you like the options of running only upgrade or both upgrade and regular tasks in a single playbook task

File: ansible_collections/arista/avd/roles/eos_designs/templates/upgrade/upgrade-2.x-to-3.0/l3leaf.j2 (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/upgrade/upgrade-2.x-to-3.0/l3leaf.j2]: fixed

**Comment by xaviramon** [comment]
LGTM!

**Review by xaviramon — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM

- [x] code review
- [x] readme review
- [x]  [validated with ipspace webinar](https://github.com/arista-netdevops-community/ipspace-webinar-september15-2020/commit/82a34c71cf628fb271b8f20aa956a2943a8e6ff7)

**Pull request merged**